internal/reader/readability/readability.go

```diff
@@ -69,12 +69,19 @@ func (c candidateList) String() string {
 }
 
 // ExtractContent returns relevant content.
-func ExtractContent(page io.Reader) (string, error) {
+func ExtractContent(page io.Reader) (string, string, error) {
 	document, err := goquery.NewDocumentFromReader(page)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
+	var baseURL string
+	document.Find("base").Each(func(i int, s *goquery.Selection) {
```

Comment:
There must be at most one `base` element, and it must be in `<head>`, so I think the query could be optimized/refined :)